services/jenkins/jenkins-build.service.js

```diff
@@ -1,107 +1,107 @@
 'use strict'
 
-const LegacyService = require('../legacy-service')
-const { makeBadgeData: getBadgeData } = require('../../lib/badge-data')
-const serverSecrets = require('../../lib/server-secrets')
+const Joi = require('joi')
+const { renderBuildStatusBadge } = require('../build-status')
+const JenkinsBase = require('./jenkins-base')
 
-// This legacy service should be rewritten to use e.g. BaseJsonService.
-//
-// Tips for rewriting:
-// https://github.com/badges/shields/blob/master/doc/rewriting-services.md
-//
-// Do not base new services on this code.
-module.exports = class JenkinsBuild extends LegacyService {
+// https://github.com/jenkinsci/jenkins/blob/master/core/src/main/java/hudson/model/BallColor.java#L56
+const schema = Joi.object({
+  color: Joi.allow(
+    'red',
+    'red_anime',
+    'yellow',
+    'yellow_anime',
+    'blue',
+    'blue_anime',
+    // green included for backwards compatibility
+    'green',
+    'green_anime',
+    'grey',
+    'grey_anime',
+    'disabled',
+    'disabled_anime',
+    'aborted',
+    'aborted_anime',
+    'notbuilt',
+    'notbuilt_anime'
```

Comment:
Instead of declaring the full list twice, you should be able to `Joi.allow(...Object.keys(colorStatusMap))` here :)

Reply:
Good call!